### PR TITLE
Finish plant search

### PR DESCRIPTION
**Pull request by badnames**
I checked the plant search use case and it seems to be fully implemented.

## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 
 ## 0.2.1 - UNRELEASED
 
-- _()_
+- Finish plant search usecase.
```
Comment (markus2330) [inline doc/CHANGELOG.md:11]: Please link to the usecase

**Review by markus2330 — CHANGES_REQUESTED** [review]
There are two left-over issues which should be mentioned in the use case:

#379 #433

**Comment by markus2330** [comment]
Please also rebase and make sure to write a manual test protocol, see #519

**Comment by markus2330** [comment]
I clarified definition of done of UC in #518

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 - CI: Added mdbook build to pipeline _(4ydan)_
 - DEV: Issue forms added. #537 _(4ydan)_
 
-## 0.2.0 - 29.06.2023
+## 0.2.0 - 27.06.2023
```
Comment (markus2330) [inline doc/CHANGELOG.md:47]: Something went wrong in the rebase.

```suggestion
## 0.2.0 - 29.06.2023
```

File: doc/CHANGELOG.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - _()_
 - _()_
-- _()_
+-
```
Comment (markus2330) [inline doc/CHANGELOG.md:36]:
```suggestion
- _()_
```

File: doc/tests/protocols/plant_search.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Test Protocol Template
+
+This is the raw template for executing a manual test.
```
Comment (markus2330) [inline doc/tests/protocols/plant_search.md:3]:
```suggestion
```

File: doc/tests/protocols/plant_search.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+|           |                                                                      |                                                                | 4. Push the search icon in the lower right menu. |                                                                                                              |               |             |                                                                                                      |
+|           |                                                                      |                                                                | 5. Write "fichte" into the search field.         |                                                                                                              |               |             |                                                                                                      |
+
+# Errors
+N/A
+
+## Closing remarks
+Some requirements from the use case have not yet been implemented in preparation for a future university course.
+Followup issues were recorded and a note has been left in the use case document.
```
Comment (markus2330) [inline doc/tests/protocols/plant_search.md:43]:
```suggestion
Follow-up issues were recorded and a note has been left in the use case document.
```

File: doc/usecases/done/search_plants.md [hunk 6]
```diff
@@ -41,3 +41,7 @@
 - **Non-functional Constraints:**
   - Performance
   - If there is a possible match in the database, it should be included (regardless of language settings etc.)
+
+# Followup
```
Comment (markus2330) [inline doc/usecases/done/search_plants.md:45]:
```suggestion
# Follow-up

```

File: doc/usecases/done/search_plants.md [hunk 7]
```diff
@@ -41,3 +41,7 @@
...
+Some parts of this use cases have not been implemented in preperation for a future university course.
+Followup issues have been created in GitHub, outlining what work still needs to be done.
```
Comment (markus2330) [inline doc/usecases/done/search_plants.md:47]: Please link these issues here and please also link the protocol. See doc/usecases/README.md for which links are expected.

```suggestion

```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thanks for the protocol, it looks nice. But not all links in the use case were created as requested in "DoD" (definition of done).

**Comment by markus2330** [comment]
> some days before a bigger release

We are hopefully a few days before :wink: If we don't start now, we never can get done next week.

> Also what do you think about the markdown table, should we maybe switch to a bullet point format

For me bullet points are easier. (Or the `- [ ]` version if something is to check).

File: doc/CHANGELOG.md [hunk 8]
```diff
@@ -10,7 +10,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 
 - Test strategy written _(4ydan)_
 - Added Nextcloud integration documentation in `/doc/research/nextcloud_integration.md`_(Samuel)_
-- _()_
+- Finish implementation of seed search _(Moritz)_
```
Comment (markus2330) [inline doc/CHANGELOG.md:13]: use random position to avoid merge conflicts (as happened now)

File: doc/usecases/done/search_plants.md [hunk 9]
```diff
@@ -41,3 +41,8 @@
 - **Non-functional Constraints:**
   - Performance
   - If there is a possible match in the database, it should be included (regardless of language settings etc.)
+
+# Follow-up
+
+Some parts of this use cases have not been implemented in preperation for a future university course.
```
Comment (markus2330) [inline doc/usecases/done/search_plants.md:47]:
```suggestion
```

**Review by markus2330 — APPROVED** [review]
Please rebase, otherwise fine. (Table looks very nice.)

**Comment by badnames** [comment]
> Thanks for creating a manual test for this use case, you did not have to perform it yet though, usually they are performed some days before a bigger release with all the existing use cases to reveal regression (maybe I need to add that to the testing strategy, so it's clearer). However, well documented. Also what do you think about the markdown table, should we maybe switch to a bullet point format or did you use some table to markdown converter for that. Because I also wrote down some tests and for productivity it wasn't the best for me.

Thanks for the feedback.

I think bullet points would make it far easier to edit the protocol.

**Comment by markus2330** [comment]
@badnames please don't self-merge. This PR broke the CI (fixed in  #629)

**Pull request merged**